FreeRTOS/FreeRTOS-Kernel
# Add coverity example

Add coverity example

Description
-----------
Add coverity example for checking MISRA violations with static analysis tool coverity.

Test Steps
-----------
The following is example command to generate coverity report:
```bash
cd ./examples/coverity

cmake -S . -B build

cd build

cov-build --emit-complementary-info --dir cov-int make

cov-analyze --dir ./cov-int --coding-standard-config ../coverity_misra.config \
--tu-pattern "file('.*/FreeRTOS/Source/[A-Za-z_]*\.c')"

cov-format-errors --dir ./cov-int --html-output html-output

```

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
